### PR TITLE
Installation instructions and quick-start tutorial

### PR DESCRIPTION
**Pull request by bartvm**
Also re-organised the TOC of the docs.

The quick-start tutorial is still incomplete (it's missing parameter intialization, monitoring, etc.) but it duplicates the unit test we had from MNIST, and it's just a nice starting point I hope.

### CODE REVIEW

**Comment by bartvm** [comment]
Build of the docs here: http://blocks.readthedocs.org/en/docs/index.html

**Comment by rizar** [comment]
I opened the link above and read this:

```
    Dimension, type and axes-checking
    Automatic creation of monitoring channels
    Easy pattern matching to select the bricks you want in large graphs
    Lazy initialization of bricks
    Application of graph transformations, such as dropout
```

We should be honest: the first two and the last one are still our plans.

**Comment by bartvm** [comment]
Forgot about those, I'll change that.

**Comment by bartvm** [comment]
Done! Since these are just docs, I'm happy to merge. Let me know if you have any objections.

File: README.rst [hunk 1]
```diff
@@ -6,19 +6,25 @@
...
-plan is to support:
+* Constructing parametrized Theano operations, called "bricks"
+* Pattern matching to select variables and bricks in large models
+* A pipeline for loading and iterating over training data
+* Algorithms to optimize your model
+* Automatic creation of monitoring channels (*limited support*)
+* Application of graph transformations, such as dropout (*limited support*)
 
-* Lazy initialization
+In the feature we also hope to support:
```
Comment (rizar) [inline README.rst]: in the future?

**Comment by rizar** [comment]
Go ahead and merge!

File: README.rst (hunk 1 above)
Comment (bartvm) [inline README.rst]: Thanks!

**Pull request merged**